test/emu_mps/test_endtoend.py

```diff
@@ -217,7 +221,82 @@ def test_XY_3atomswith_slm() -> None:
     )  # todo, compare against pulser results
 
 
-def test_end_to_end_afm_ring() -> None:
+@pytest.mark.parametrize(
+    "optimize_order",
+    [
+        False,
+        True,
+    ],
+)
+def test_end_to_end_domain_wall_ring(
+    optimize_order: bool,
+) -> None:
+    torch.manual_seed(seed)
+
+    num_qubits = 6
+    seq = pulser_afm_sequence_ring(
+        num_qubits=num_qubits,
+        Omega_max=0,
+        U=U,
+        delta_0=delta_0,
+        delta_f=delta_f,
+        t_rise=1300,
+        t_fall=1400,
+    )
+
+    initial_state = emu_mps.MPS.from_state_amplitudes(
+        eigenstates=("r", "g"),
+        amplitudes={(num_qubits // 2) * "r" + (num_qubits // 2) * "g": 1.0},
+    )
+
+    # I want to test permutation results close to the initial state
+    eval_times = [1 / 44, 1]  # 1/44 is 1 dt step
+    mps_config = MPSConfig(
+        initial_state=initial_state,
+        dt=100,
+        precision=1e-5,
+        observables=[
+            BitStrings(evaluation_times=eval_times, num_shots=100),
+            # Fidelity(evaluation_times=eval_times, state=fidelity_state, tag_suffix="1"),
+            Occupation(evaluation_times=eval_times),
+            Energy(evaluation_times=eval_times),
+            EnergyVariance(evaluation_times=eval_times),
+            CorrelationMatrix(evaluation_times=eval_times),
+        ],
+        optimize_qubit_ordering=optimize_order,
+    )
+
+    backend = MPSBackend(seq, config=mps_config)
+    result = backend.run()
+
+    ntime_step = 0
+    bitstrings = result.bitstrings[ntime_step]
+    occupation = result.occupation[ntime_step]
+    cor_mat = result.correlation_matrix[ntime_step]
+    energy = result.energy[ntime_step]
+    energy_variance = result.energy_variance[ntime_step]
+
+    expect_occup = torch.tensor([1, 1, 1, 0, 0, 0], dtype=torch.float64)
+    expect_corr = torch.outer(expect_occup, expect_occup).to(torch.complex128)
+
+    assert bitstrings["111000"] == 100
```

Comment:
since I asked you to also fill in the `qubit_order` in `Results` perhaps also add an assert on the order in this test.

Reply:
done in the other test 'domain_wall'